packages/playwright-test/src/runner/projectUtils.ts

```diff
@@ -61,21 +61,27 @@ export function buildTeardownToSetupsMap(projects: FullProjectInternal[]): Map<F
   return result;
 }
 
-export function buildProjectsClosure(projects: FullProjectInternal[]): Map<FullProjectInternal, 'top-level' | 'dependency'> {
+export function buildProjectsClosure(projects: FullProjectInternal[], hasTests?: (project: FullProjectInternal) => boolean): Map<FullProjectInternal, 'top-level' | 'dependency'> {
```

Comment:
Please check that resulting map is not iterated by projects that might be missing from it.